src/distilabel/distiset.py

```diff
@@ -193,6 +225,193 @@ def train_test_split(
+
+        fs: fsspec.AbstractFileSystem
+        fs, _, _ = fsspec.get_fs_token_paths(
+            distiset_config_folder, storage_options=storage_options
+        )
+        fs.makedirs(distiset_config_folder, exist_ok=True)
+
+        if save_card:
+            # NOTE: Currently the card is not the same if we write to disk or push to the HF hub,
+            # as we aren't generating the README copying/updating the data from the dataset repo.
+            card = self._get_card(repo_id=Path(distiset_path).stem, token=None)
+            new_filename = posixpath.join(distiset_config_folder, "README.md")
+            if storage_options:
+                # Write the card the same way as DatasetCard.save does:
+                with fs.open(new_filename, "w", newline="", encoding="utf-8") as f:
+                    f.write(str(card))
+            else:
+                card.save(new_filename)
+
+        # Write our internal files to the distiset folder by copying them to the distiset folder.
+        if save_pipeline_config and self.pipeline_path:
+            new_filename = posixpath.join(
+                distiset_config_folder, PIPELINE_CONFIG_FILENAME
+            )
+            if self.pipeline_path.exists() and (not fs.isfile(new_filename)):
+                data = yaml.safe_load(
+                    self.pipeline_path.read_text(), Loader=yaml.FullLoader
+                )
+                with fs.open(new_filename, "w", encoding="utf-8") as f:
+                    yaml.dump(data, f, default_flow_style=False)
+
+        if save_pipeline_log and self.log_filename_path:
+            new_filename = posixpath.join(distiset_config_folder, PIPELINE_LOG_FILENAME)
+            if self.log_filename_path.exists() and (not fs.isfile(new_filename)):
+                data = self.log_filename_path.read_text()
+                with fs.open(new_filename, "w", encoding="utf-8") as f:
+                    f.write(data)
+
+    @classmethod
+    def load_from_disk(
+        cls,
+        distiset_path: PathLike,
+        keep_in_memory: Optional[bool] = None,
+        storage_options: Optional[Dict[str, Any]] = None,
+    ) -> Self:
+        """Loads a dataset that was previously saved using `Distiset.save_to_disk` from a dataset
+        directory, or from a filesystem using any implementation of `fsspec.spec.AbstractFileSystem`.
+
+        Args:
+            distiset_path (PathLike): Path ("dataset/train") or remote URI ("s3://bucket/dataset/train").
+            keep_in_memory: Whether to copy the dataset in-memory, see `datasets.Dataset.load_from_disk``
+                for more information. Defaults to None.
+            storage_options: Key/value pairs to be passed on to the file-system backend, if any.
+                Defaults to None.
+
+        Returns:
+            Distiset
+        """
+        distiset_path = str(distiset_path)
+        fs: fsspec.AbstractFileSystem
+        fs, _, [distiset_path] = fsspec.get_fs_token_paths(
```

Comment:
I think, I found an issue:
when loading with a `distiset_path` that has `"s3://"`as prefix. This prefix gets discarded here.
This line turns `"s3://my/example/dataset"` into `"my/example/dataset"` and subsequent loading from S3 with `Dataset.load_from_disk()` fails. (Effectively it interprets the truncated path as an local path and fails, if no such local dataset is present.)

Reply:
I'm currently looking into, if this can be solved via an option in `storage_options`

Reply:
![Screenshot 2024-05-27 at 16 29 36](https://github.com/argilla-io/distilabel/assets/73563550/99347ebd-c9b8-4415-9fa2-6302881ab46e)

The paths are missing `"s3://"` and start right away with `"example/path"`

Reply:
It should be fixed now @rasdani, but if you have time let me know if it works for you 😄

Reply:
@plaguss 
yes, I can load the Distiset with `"/default"` at the end of my path :)
I think this matches `datasets` behaviour for `DatasetDict`s.

Reply:
after saving once more with `Distitset.save_to_disk()` to S3, I no don't need `"/default"` at the end when loading.
Not 100% sure how `datasets` handles this, but this works for us now 👍

Reply:
Thanks for the confirmation @rasdani, indeed Hugging Face handles that automatically whenever the config is `default` i.e. only one config within the dataset; and when loading the dataset and multiple configs are available all the configs are downloaded and loaded. Alternatively one can also provide the `config_name` to load one config or the other. But long-story-short, the behaviour now is as intended, the `/default` or whatever config name there is, should not be provided. 🤗